File: vortex_ops/vortex_ops/vortex_ops/report/loan_balance_ledger/loan_balance_ledger.js
frappe.query_reports["Loan Balance Ledger"] = {
    filters: [
        {
            fieldname: "streamer",
            fieldtype: "Link",
            label:     "Streamer",
            options:   "Streamer",
        },
        {
            fieldname: "status",
            fieldtype: "Select",
            label:     "Status",
            options:   "\nActive\nPaid Off\nForgiven\nOn Hold\nCancelled",
        },
    ],

    formatter(value, row, column, data, default_formatter) {
        value = default_formatter(value, row, column, data);
        if (!data) return value;

        if (data._is_total) {
            value = `<strong>${value}</strong>`;
        }

        if (column.fieldname === "status" && data.status) {
            const colors = {
                "Active":    "#16a34a",
                "Paid Off":  "#6b7280",
                "Forgiven":  "#9333ea",
                "On Hold":   "#d97706",
                "Cancelled": "#dc2626",
            };
            const c = colors[data.status];
            if (c) return `<span style="color:${c}; font-weight:600;">${data.status}</span>`;
        }

        // Highlight outstanding balance in amber
        if (column.fieldname === "balance" && (data.balance || 0) > 0) {
            return `<span style="color:#d97706; font-weight:600;">${value}</span>`;
        }

        return value;
    },
};

File: vortex_ops/vortex_ops/vortex_ops/report/loan_balance_ledger/loan_balance_ledger.py
import frappe
from vortex_ops.utils import safe_float


def execute(filters=None):
    f    = filters or {}
    cond = ["rec.docstatus = 1"]
    vals = []

    if f.get("streamer"):
        cond.append("rec.streamer = %s")
        vals.append(f["streamer"])
    if f.get("status"):
        cond.append("rec.status = %s")
        vals.append(f["status"])

    rows = frappe.db.sql(
        f"""
        SELECT
            rec.name          AS loan_id,
            rec.streamer      AS streamer,
            rec.loan_date     AS loan_date,
            rec.loan_amount   AS loan_amount,
            rec.amount_repaid AS amount_repaid,
            rec.balance       AS balance,
            rec.status        AS status,
            rec.notes         AS notes
        FROM `tabLoan Record` rec
        WHERE {" AND ".join(cond)}
        ORDER BY rec.streamer, rec.loan_date
        """,
        vals,
        as_dict=True,
    )

    data        = list(rows)
    grand_total = round(sum(safe_float(r.get("balance")) for r in data), 2)

    if data:
        data.append({
            "streamer":  "TOTAL OUTSTANDING",
            "balance":   grand_total,
            "_is_total": True,
        })

    return _columns(), data


def _columns():
    return [
        {"label": "Loan ID",     "fieldname": "loan_id",       "fieldtype": "Link",
         "options": "Loan Record", "width": 160},
        {"label": "Streamer",    "fieldname": "streamer",      "fieldtype": "Link",
         "options": "Streamer",    "width": 140},
        {"label": "Loan Date",   "fieldname": "loan_date",     "fieldtype": "Date",     "width": 100},
        {"label": "Loan Amount", "fieldname": "loan_amount",   "fieldtype": "Currency", "width": 120},
        {"label": "Repaid",      "fieldname": "amount_repaid", "fieldtype": "Currency", "width": 110},
        {"label": "Balance",     "fieldname": "balance",       "fieldtype": "Currency", "width": 110},
        {"label": "Status",      "fieldname": "status",        "fieldtype": "Data",     "width": 100},
        {"label": "Notes",       "fieldname": "notes",         "fieldtype": "Data",     "width": 200},
    ]
